source/monitoring-logging/logging.txt

```diff
@@ -0,0 +1,195 @@
+- Message: Describes the logged event.
+- Domain string: Specifies the driver component that emitted the log message.
+
+To implement ``Psr\Log\LoggerInterface``, you can create a class that
+extends the ``Psr\Log\AbstractLogger`` class. ``AbstractLogger`` implements
+``LoggerInterface`` and a generic ``log()`` method, which receives log
+messages at each severity level.
+
+Example
+~~~~~~~
+
+This example performs the following actions:
+
+- Creates a logger called ``MyLogger`` that extends the ``AbstractLogger`` class
+  and records the log level, description, and domain of each event
+- Registers the logger
+- Creates a client that connects to a CosmosDB cluster
+- Prints each log message
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/monitoring-logging/logging.php
+      :start-after: start-register-logger
+      :end-before: end-register-logger
+      :language: php
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      Array
+      (
+          [0] => Array
+              (
+                  [0] => debug
+                  [1] => Connection string: 'mongodb://a.mongo.cosmos.azure.com:19555/'
+                  [2] => PHONGO
+              )
+          [1] => Array
+              (
+                  [0] => debug
+                  [1] => Creating Manager, phongo-1.21.0[stable] - mongoc-1.30.1(bundled), libbson-1.30.1(bundled), php-8.4.4
+                  [2] => PHONGO
+              )
+          [2] => Array
+              (
+                  [0] => debug
+                  [1] => Setting driver handshake data: { name: 'ext-mongodb:PHP / PHPLIB ', version: '1.21.0 / 1.21.1 ', platform: 'PHP 8.4.4 ' }
+                  [2] => PHONGO
+              )
+          [3] => Array
+              (
+                  [0] => info
+                  [1] => You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb
+                  [2] => mongoc
+              )
+          [4] => Array
+              (
+                  [0] => debug
+                  [1] => Created client with hash: a:4:{s:3:"pid";i:61731;s:3:"uri";s:41:"mongodb://a.mongo.cosmos.azure.com:19555/";s:7:"options";a:0:{}s:13:"driverOptions";a:1:{s:6:"driver";a:2:{s:4:"name";s:6:"PHPLIB";s:7:"version";s:6:"1.21.1";}}}
+                  [2] => PHONGO
+              )
+          [5] => Array
+              (
+                  [0] => debug
+                  [1] => Stored persistent client with hash: a:4:{s:3:"pid";i:61731;s:3:"uri";s:41:"mongodb://a.mongo.cosmos.azure.com:19555/";s:7:"options";a:0:{}s:13:"driverOptions";a:1:{s:6:"driver";a:2:{s:4:"name";s:6:"PHPLIB";s:7:"version";s:6:"1.21.1";}}}
+                  [2] => PHONGO
+              )
+      )
+
+The preceding output includes log messages with a ``debug`` severity level, which
+describe normal driver activities. It also includes a message about the CosmosDB
```

Comment:
S: to resolve the vale issue you could use a phrase like "These messages provide updates on the state of the connection" as normal is subjective